src/item.c

```diff
@@ -0,0 +1,1046 @@
+    default:
+        return 0;
+    }
+}
+
+u16 TMHMGetMove(u16 itemId) {
+    if (itemId < ITEM_TM01 || itemId > ITEM_HM08) {
+        return MOVE_NONE;
+    }
+
+    itemId -= ITEM_TM01;
+    return _021000CC[itemId];
+}
+
+BOOL MoveIsHM(u16 moveId) {
+    u8 i;
+
+    for (i = 0; i < NUM_HMS; i++) {
+        if (_021000CC[i + ITEM_HM01 - ITEM_TM01] == moveId) {
+            return TRUE;
+        }
+    }
+
+    return FALSE;
+}
+
+u8 ItemToTMHMId(u16 itemId) {
+    if (itemId < ITEM_TM01 || itemId > ITEM_HM08) {
+        return 0;
+    }
+
+    return itemId - ITEM_TM01;
+}
+
+BOOL ItemIdIsMail(u16 itemId) {
+    if (itemId < FIRST_MAIL_IDX || itemId > LAST_MAIL_IDX) {
+        return FALSE;
+    } else {
+        return TRUE;
+    }
+}
+
+u8 ItemToMailId(u16 itemId) {
+    if (itemId < FIRST_MAIL_IDX || itemId > LAST_MAIL_IDX) {
+        return 0;
+    } else {
+        return itemId - ITEM_GRASS_MAIL + MAIL_GRASS;
+    }
+}
+
+u16 MailToItemId(u8 mailId) {
+    if (mailId >= NUM_MAIL) {
+        return ITEM_NONE;
+    } else {
+        return mailId - MAIL_GRASS + ITEM_GRASS_MAIL;
+    }
+}
+
+s32 ItemToBallId(u16 itemId) {
+    if (itemId >= ITEM_MASTER_BALL && itemId <= ITEM_CHERISH_BALL) {
+        return itemId - ITEM_MASTER_BALL + BALL_MASTER;
+    } else if (itemId >= ITEM_FAST_BALL && itemId <= ITEM_SPORT_BALL) {
+        return itemId - ITEM_FAST_BALL + BALL_FAST;
+    } else {
+        return BALL_POKE;
+    }
+}
+
+u16 BallToItemId(s32 ballId) {
+    if (ballId >= BALL_MASTER && ballId <= BALL_CHERISH) {
+        return ballId - BALL_MASTER + ITEM_MASTER_BALL;
+    } else if (ballId >= BALL_FAST && ballId <= BALL_SPORT) {
+        return ballId - BALL_FAST + ITEM_FAST_BALL;
+    } else {
+        return ITEM_POKE_BALL;
+    }
+}
+
+BOOL ItemIdIsBerry(u16 itemId) {
+    if (itemId < FIRST_BERRY_IDX || itemId > LAST_BERRY_IDX) {
+        return FALSE;
+    } else {
+        return TRUE;
+    }
+}
+
+u16 BerryToItemId(u8 berryId) {
+    if (berryId >= NUM_BERRIES)
+        return 0xFFFF;
+    return (u16)(berryId + FIRST_BERRY_IDX);
+}
+
+STRING *GetNutName(u16 berryId, u32 heap_id) {
+    STRING *ret;
+    MSGDATA *msgData;
+
+    if (berryId != 0) {
+        berryId--;
+    }
+    msgData = NewMsgDataFromNarc(MSGDATA_LOAD_LAZY, NARC_msgdata_msg, NARC_msg_msg_00000251_bin, heap_id);
+    ret = NewString_ReadMsgData(msgData, berryId);
+    DestroyMsgData(msgData);
+    return ret;
+}
+
+ITEMDATA *LoadAllItemData(u32 heap_id) {
+    return AllocAndReadFromNarcMemberByIdPair(NARC_itemtool_itemdata_item_data, 0, heap_id, 0, GetItemIndexMapping(ITEM_MAX, ITEMNARC_PARAM) * sizeof(ITEMDATA));
+}
+
+ITEMDATA *GetItemDataPtrFromArray(ITEMDATA *itemData, u32 itemDataIdx) {
+    return &itemData[itemDataIdx];
+}
+
+BOOL ItemIdIsNotJohtoBall(u16 itemId) {
+    // stupid hack to get it to match
+    if ((u16)(itemId + (u16)-ITEM_FAST_BALL) > (ITEM_PARK_BALL - ITEM_FAST_BALL)) {
+        return TRUE;
+    } else {
+        return FALSE;
+    }
```

Comment:
```suggestion
    return ((u16)(itemId + (u16)-ITEM_FAST_BALL) > (ITEM_PARK_BALL - ITEM_FAST_BALL));
```

Reply:
Rejecting change to maintain consistency with other files